### PR TITLE
docs: OpenAIInstrumentor tutorial notebook

### PR DESCRIPTION
**Pull request by axiomofjoy**
- demonstrate structured data extraction using OpenAI function calling
- demonstrate `OpenAIInstrumentor`

Follow-up: [Add evals for structured data extraction](https://github.com/Arize-ai/phoenix/issues/1581)

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ repos:
             files: \.ipynb$
             stages: [commit]
             language: system
-            entry: nbqa ruff --fix --ignore=E402
+            entry: nbqa ruff --fix --ignore=E402 --ignore=E501
```
Comment (axiomofjoy) [inline .pre-commit-config.yaml:27]: Ignore max line length for notebooks.

File: tutorials/tracing/openai_tracing_tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+    "Instrument your OpenAI client with a tracer that emits telemetry data in OpenInference format. [OpenInference](https://arize-ai.github.io/open-inference-spec/trace/) is an open standard for capturing and storing LLM application traces that enables LLM applications to seamlessly integrate with LLM observability solutions such as Phoenix."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tracer = Tracer(exporter=HttpExporter())\n",
```
Comment (mikeldking) [inline tutorials/tracing/openai_tracing_tutorial.ipynb:126]: Might just nix this to keep it simple

**Comment by mikeldking** [comment]
Very neat 👏

**Review by mikeldking — APPROVED** [review]

File: tutorials/tracing/openai_tracing_tutorial.ipynb (hunk 2 above)
Comment (axiomofjoy) [reply tutorials/tracing/openai_tracing_tutorial.ipynb:126]: By default, our tracer does not have an exporter. We would need to change the default to `HttpExporter` in order for traces to show up in Phoenix. https://github.com/Arize-ai/phoenix/blob/5128abe0add224c506aba7cce3b06389818eb09c/src/phoenix/trace/tracer.py#L57

**Pull request merged**